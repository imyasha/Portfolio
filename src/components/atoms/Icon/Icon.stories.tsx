import { Icon, IconProps } from './Icon';

export default {
	title: 'atoms/Icon',
	component: Icon,
};

export const IconStory = (args: IconProps) => <Icon {...args} />;

IconStory.storyName = 'Icon';
IconStory.args = {
	icon: 'GITHUB',
};
IconStory.argTypes = {
	icon: {
		control: {
			type: 'select',
			options: ['GITHUB', 'LINKEDIN', 'INSTAGRAM', 'TWITTER'],
		},
	},
};
IconStory.decorators = [
	(Story: any) => (
		<div style={{ width: 32, height: 32 }}>
			<Story />
		</div>
	),
];
