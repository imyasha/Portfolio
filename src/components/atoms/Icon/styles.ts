import styled from "styled-components"

const StyledIcon = styled.svg``;

export { StyledIcon };
