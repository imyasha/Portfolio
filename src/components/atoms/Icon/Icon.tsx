import { FC } from 'react';
import { StyledIcon } from './styles';

import Github from 'Public/assets/github.svg';
import Linkedin from 'Public/assets/linkedin.svg';
import Instagram from 'Public/assets/instagram.svg';
import Twitter from 'Public/assets/twitter.svg';

export interface IconProps {
	icon: 'GITHUB' | 'LINKEDIN' | 'INSTAGRAM' | 'TWITTER';
}

const Icon: FC<IconProps> = ({ icon }) => {
	let Icon = () => <svg />;

	switch (icon) {
		case 'GITHUB':
			Icon = Github;
			break;
		case 'LINKEDIN':
			Icon = Linkedin;
			break;
		case 'INSTAGRAM':
			Icon = Instagram;
			break;
		case 'TWITTER':
			Icon = Twitter;
			break;
		default:
			break;
	}

	return (
		<>
			<Icon />
		</>
	);
};

export { Icon };
