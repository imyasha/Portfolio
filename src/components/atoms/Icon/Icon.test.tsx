import { Icon } from '../Icon';
import { render } from '@testing-library/react';

const icons = ['GITHUB', 'LINKEDIN', 'INSTAGRAM', 'TWITTER'];

describe('Icon', () => {
	icons.forEach((icon) => {
		it(`renders ${icon} correctly`, () => {
			// @ts-ignore
			const { container } = render(<Icon icon={icon} />);
			expect(container).toMatchSnapshot();
		});
	});
});
