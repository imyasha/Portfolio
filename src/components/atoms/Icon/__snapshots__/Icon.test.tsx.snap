// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders GITHUB correctly 1`] = `
<div>
  <test-file-stub />
</div>
`;

exports[`Icon renders INSTAGRAM correctly 1`] = `
<div>
  <test-file-stub />
</div>
`;

exports[`Icon renders LINKEDIN correctly 1`] = `
<div>
  <test-file-stub />
</div>
`;

exports[`Icon renders TWITTER correctly 1`] = `
<div>
  <test-file-stub />
</div>
`;
